src/ai/backend/manager/models/endpoint.py

```diff
@@ -631,6 +645,87 @@ class Meta:
     items = graphene.List(Endpoint, required=True)
 
 
+class ModifyEndpointInput(graphene.InputObjectType):
+    resource_slots = graphene.JSONString()
+    resource_opts = graphene.JSONString()
+    cluster_mode = graphene.String()
+    cluster_size = graphene.Int()
+    desired_session_count = graphene.Int()
+    image = ImageRefType()
+    name = graphene.String()
+    resource_group = graphene.String()
+    open_to_public = graphene.Boolean()
+
+
+class ModifyEndpoint(graphene.Mutation):
```

Comment:
There is no permission evaluation at all for this mutation - that means unprivileged users can use this mutation to update endpoint information of others. Please update the endpoint to include permission measurements.